test/t_aes_ecb.cc

```diff
@@ -184,6 +184,99 @@ TEST(aes_ecb, encrypt)
 	EXPECT_EQ(aes_key, nullptr);
 }
 
+TEST(aes_ecb, encrypt_session_bound)
+{
+	struct zpc_aes_key *aes_key = NULL, *aes_key2 = NULL;
+	struct zpc_aes_ecb *aes_ecb = NULL;
+	const char *mkvp, *apqns[257];
+	u8 m[64], c[64];
+	unsigned int flags;
+	int rc, size, type;
+	u8 buf[10000];
+	size_t buflen = sizeof(buf);
+
+	TESTLIB_ENV_AES_KEY_CHECK();
+
+	TESTLIB_AES_ECB_HW_CAPS_CHECK();
+
+	TESTLIB_AES_KERNEL_CAPS_CHECK();
+
+	size = testlib_env_aes_key_size();
+	type = testlib_env_aes_key_type();
+	flags = testlib_env_aes_key_flags();
+	mkvp = testlib_env_aes_key_mkvp();
+	(void)testlib_env_aes_key_apqns(apqns);
+
+	if (type != ZPC_AES_KEY_TYPE_EP11)
+		GTEST_SKIP_("Skipping encrypt_session_bound test. Only applicable for EP11 type keys.");
+
+	TESTLIB_AES_SW_CAPS_CHECK(type);
+
+	rc = zpc_aes_key_alloc(&aes_key);
+	EXPECT_EQ(rc, 0);
+	rc = zpc_aes_key_alloc(&aes_key2);
+	EXPECT_EQ(rc, 0);
+	rc = zpc_aes_ecb_alloc(&aes_ecb);
+	EXPECT_EQ(rc, 0);
+
+	rc = zpc_aes_key_set_type(aes_key, type);
+	EXPECT_EQ(rc, 0);
+	rc = zpc_aes_key_set_type(aes_key2, type);
+	EXPECT_EQ(rc, 0);
+	if (mkvp != NULL) {
+		rc = zpc_aes_key_set_mkvp(aes_key, mkvp);
+		EXPECT_EQ(rc, 0);
+		rc = zpc_aes_key_set_mkvp(aes_key2, mkvp);
+		EXPECT_EQ(rc, 0);
+	} else {
+		rc = zpc_aes_key_set_apqns(aes_key, apqns);
+		EXPECT_EQ(rc, 0);
+		rc = zpc_aes_key_set_apqns(aes_key2, apqns);
+		EXPECT_EQ(rc, 0);
+	}
+	rc = zpc_aes_key_set_size(aes_key, size);
+	EXPECT_EQ(rc, 0);
+	rc = zpc_aes_key_set_size(aes_key2, size);
+	EXPECT_EQ(rc, 0);
+	rc = zpc_aes_key_set_flags(aes_key, flags);
+	EXPECT_EQ(rc, 0);
+	rc = zpc_aes_key_set_flags(aes_key2, flags);
+	EXPECT_EQ(rc, 0);
+	rc = zpc_aes_key_generate(aes_key);
+	EXPECT_EQ(rc, 0);
+
+	/*
+	 * Now export this key into a buffer, give it a fake session id, and
+	 * re-import it again.
+	 */
+	rc = zpc_aes_key_export(aes_key, buf, &buflen);
+	EXPECT_EQ(rc, 0);
+	memset(buf + 16, 0x5c, 32);
```

Comment:
Hm, using this key win in any case fail.
EP11 secure keys are MACed, so if you place a session id into it the way you do it, the MAC is no longer valid, and the crypto card will reject to use that key. 

You essentially have no easy way of generating a session bound EP11 key blob that really works. You would need to use EP11 host lib calls like m_Login to establish a session and then generate a key using m_GenerateKey to get a session bound key blob.

So unless you can distinguish between the case where the secure to protected key conversion fails because of the missing kernel support, and the case where the conversion fails because the key's MAC is not valid, this test is completely useless and will always fail.

Reply:
Hmm yes, I didn't have the MAC in mind ... so yes, we can leave this test out. I will remove this commit completely.

Reply:
... and the new macro TESTLIB_INFO is then also no more needed.